docs/en/ingest-management/fleet-agent-proxy-support.asciidoc

```diff
@@ -0,0 +1,252 @@
+
+* Download artifacts from `artifacts.elastic.co` for subprocesses or binary
+upgrades
+* Send data to {es}
+* Retrieve agent policies from {fleet-server}
+* Retrieve agent policies from {es} (only needed for agents running {fleet-server})
+* Monitor services (not shown)
+
+image::images/agent-proxy-server.png[Image showing connections between {agent}, {fleet-server}, and {es}]
+
+You can configure {agent} proxy settings by specifying:
+
+* <<host-proxy-env-vars>>
+* <<proxy-settings-in-agent-policy>>
+* <<cli-proxy-settings>>
+
+[discrete]
+[[host-proxy-env-vars]]
+=== Host environment variables
+
+Set environment variables on the host to configure default proxy settings.
+The {agent} uses host environment settings by default if no proxy settings are
+specified elsewhere. You can override host proxy settings later when you
+configure the {agent} and {fleet} settings. The following environment variables
+are available on the host:
+
+|===
+|Variable |Description
+
+|`HTTP_PROXY`
+|URL of the proxy server for HTTP traffic. 
+
+|`HTTPS_PROXY`
+|URL of the proxy server for HTTPS traffic.
+
+|`NO_PROXY`
+|IP addresses or domain names that should not use the proxy. Supports patterns.
+|===
+
+The proxy URL can be a complete URL or `host[:port]`, in which case the `http`
+scheme is assumed. An error is returned if the value is a different form.
+
+Keep in mind that proxy environment settings may affect other components on the
+host that send HTTP requests.
+
+//TODO: Add examples of how/where to set env vars on various platforms.
+
+[discrete]
+[[proxy-settings-in-agent-policy]]
+=== Proxy settings in the {agent} policy
+
+Proxy settings in the {agent} policy override proxy settings specified by
+environment variables. The following proxy settings are valid in the agent
+policy:
+
+|===
+|Setting | Description
+
+|`proxy_url`
+| (string) URL of the proxy server. If set, the configured URL is used as a
+proxy for all connection attempts by the component. The value may be either a
+complete URL or a `host[:port]`, in which case the `http` scheme is assumed. If
+a value is not specified through the configuration, then proxy environment
+variables are used. The URL accepts optional `username` and `password` settings
+for authenticating with the proxy. For example:
+`http://<username>:<password>@<proxy host>/`.
+
+|`proxy_headers`
+| (string) Additional headers to send to the proxy during CONNECT requests. You
+can use this setting to pass keys/tokens required for authenticating with the
+proxy.
+
+|`proxy_disable`
+| (boolean) If set to `true`, all proxy settings, including the `HTTP_PROXY` and
+`HTTPS_PROXY` environment variables, are ignored.
+
+|===
+
+Proxy settings are valid in the {agent} policy under the following sections:
+
+* `outputs` - set the proxy server for communicating with {es}.
+
+* `fleet` - set the proxy server for retrieving agent policies from
+{fleet}.
+
+* `agent.download` - set the proxy server for retrieving artifacts or binary
```

Comment:
@michel-laterman One follow-up question about your comment: "...(agent.download.proxy_url) needs to be injected into fleet.yml when in fleet mode." When you say "needs to be injected into fleet.yml," do you mean that users need to modify that file manually? What happens if they update the policy if Fleet after adding this setting? Is the setting preserved, or do they need to go back and set it again?

Reply:
Yes, the user needs to manually add it to the file.
It does not get removed when policies update/the agent restarts
I've tested restarting the elastic-agent service (`systemctl restart elastic-agent.service`), calling `elastic-agent restart`, and updating the policy;

Reply:
Fleet is supposed to provide a way to centrally manage agents, which means users should not have to modify files manually on each host. I see this file more like an internal file than something users edit directly. It seems like the Kibana UI would provide a better solution when its available https://github.com/elastic/kibana/issues/100413. However, that means the future UI may override these user settings. I'm a little uncomfortable documenting this if it could lead to breaking changes later. If that's the case, we could defer documenting this until we have a more fully designed solution. What do you think @joshdover?

Reply:
I believe a workaround is that users could set a proxy server via environment variable. The question is here is whether we want to allow users to set a different proxy server for the downloads from other connections, which I think is a less common use case.

Reply:
> 

It would be beneficial to my situation  if I could set a different proxy value  for this,  different to some of the other proxy settings I use/set in other areas of  the yml.   

As long as the documentation is clear on its intention people will be happy.

Reply:
I prefer to document this, along with the caveat that the setting might get overridden in the future. IMO it will save users and the support team some effort in figuring out how to do this until we get to the point where it's properly supported in the UI.

Reply:
I'm going to mark this as closed, and you can comment on the commit after I push it.